File: Algorithm/Sort/InsertionSort/InsertionSort.cpp
#include <iostream>
using namespace std;

void outputArray(int arr[],int len) {
    while (len-- >0) {
        printf("%d ",*arr++);
    }
    printf("\n");
}

void insetion_sort(int arr[], int len) {
    for (int i = 1, j= 0; i < len; ++i) {
        int temp = arr[i];
        for (j = i - 1; j >= 0 && arr[j] > temp; --j) {
            arr[j + 1] = arr[j];
        }
        arr[j + 1] = temp;
    }
}

int main(int argc, char*argv[]) {
    int a[] = {5, 9, 20, 31, 8, 89, 1, 15, 100, 2};
    cout<<"before sorting"<<endl;
    outputArray(a, 10);
    cout<<"after sorting"<<endl;
    insetion_sort(a,10);
    outputArray(a, 10);

    return 0;
}

File: Algorithm/Sort/InsertionSort/InsertionSort.py
#!/usr/local/bin/python3
def insertion_sort(lst):
    if len(lst) == 1:
        return lst

    for i in range(1, len(lst)):
        temp = lst[i]
        j = i - 1
        while j >= 0 and temp < lst[j]:
            lst[j + 1] = lst[j]
            j -= 1
        lst[j + 1] = temp
    return lst


testlist = [17, 23, 20, 14, 12, 25, 1, 20, 81, 14, 11, 12]
print('Before selection sort: {}'.format(testlist))
print('After selection sort:  {}'.format(insertion_sort(testlist)))
